### PR TITLE
Level Detective Processor - Audio Processing & Visualiser

### PR DESCRIPTION
**Pull request by RachelMaryamLocke**
Screenshot 2023-07-25 at 00 51 08

### CODE REVIEW

File: src/processors/other/LevelDetective.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "../../modules/chowdsp_utils/modules/gui/chowdsp_visualizers/CompressorPlots/chowdsp_LevelDetectorVisualizer.h"
```
Comment (jatinchowdhury18) [inline src/processors/other/LevelDetective.h:3]: I don't think we should need this include? `<chowdsp_visualizers/chowdsp_visualizers.h>` should be included in JuceHeader.h, and which should have this header included internally.

**Comment by jatinchowdhury18** [comment]
Looking good overall! I was thinking it might make sense to override `processAudioBypassed()`. In its current form, I think the bypass behaviour would just pass through the audio without going through the level detector. I think more likely the behaviour we would want would be a "silent" output if the processor bypassed. Would you agree?

**Comment by RachelMaryamLocke** [comment]
Yes, I agree, it wouldn't make sense for the level output to send out mono audio information when the processor is bypassed, so in the case it is bypassed I have cleared the levelOutBuffer.

**Comment by RachelMaryamLocke** [comment]
Let me know if I'm ok to merge this :)

File: src/processors/other/LevelDetective.cpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            for (auto* s : { &attackSlider, &releaseSlider })
+            {
+                s->setSliderStyle (Slider::RotaryHorizontalVerticalDrag);
+                s->setTextBoxStyle (Slider::TextBoxBelow, false, 80, 20);
+                s->setColour (Slider::textBoxHighlightColourId, powerColour.withAlpha (0.55f));
+                s->setColour (Slider::thumbColourId, powerColour);
+                addAndMakeVisible (s);
+            }
+
+            visualiser.backgroundColour = backgroundColour.darker (0.4f);
```
Comment (jatinchowdhury18) [inline src/processors/other/LevelDetective.cpp:128]: So to be able to push this change through, we'll need to update the submodule for chowdsp_utils. So the steps for that are:
- in your local copy create new branch and make a PR with your changes
- once the PR is merged, pull the submodule to the latest commit, and then push the submodule update to this repo

File: src/processors/other/LevelDetective.cpp [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    }
+    else
+    {
+        levelOutBuffer.clear();
+    }
+
+    outputBuffers.getReference (LevelOutput) = &levelOutBuffer;
+}
+
+void LevelDetective::processAudioBypassed (AudioBuffer<float>& buffer)
```
Comment (jatinchowdhury18) [inline src/processors/other/LevelDetective.cpp:86]: At the moment the visualizer doesn't update when the module is bypassed... do you think that's the correct behaviour? I'm kind of on the fence.

File: src/processors/other/LevelDetective.cpp (hunk 2 above)
Comment (RachelMaryamLocke) [reply src/processors/other/LevelDetective.cpp:128]: Is this different to the branch and PR I created for utils with the changes to waveformView? [https://github.com/Chowdhury-DSP/chowdsp_utils/pull/446]

Comment (jatinchowdhury18) [reply src/processors/other/LevelDetective.cpp:128]: Nope that's the one! Just merged it.

**Pull request merged**